### DISCUSSION TITLE
cypress e2e cy.request() failed trying to load http://localhost:9001/ Error: socket hang up

### DISCUSSION THREAD

**Discussion by pmatsibekker** [cypress-io/cypress#25734] [Questions and Help]
I am able to open electron browser directly to my webpack dev server ui on port 9001 but cypress is failing on 
cy.request('http://localhost:9001/',{timeout:10000})

Seems like it doesn't wait the 10 seconds and just fails close to immediately.
I am able to run to other hosts and even connect to other rest controller ports on localhost...just seems to be issue with webpack dev server connect.

**Comment by portyu9** [comment]
Hi @pmatsibekker 

There are two separate things here.

First, this does **not** set a 10-second timeout:

```js
cy.request('http://localhost:9001/', { timeout: 10000 })
```

With that overload, the second argument is interpreted as the **request body**.

Use the options-object form instead:

```js
cy.request({
  url: 'http://localhost:9001/',
  timeout: 10000,
})
```

That explains why your timeout setting appeared to have no effect.

It does not, however, prove why the connection is getting `socket hang up`.

An important difference is that `cy.request()` is sent from the **Cypress Node process**, not from Electron/the browser. Therefore “I can open it successfully in Electron” does not prove the same network path works for `cy.request()`.

If the corrected call still gets `socket hang up`, check from the same machine/container where Cypress runs:

```bash
curl -v http://localhost:9001/
```

Also try:

```text
http://127.0.0.1:9001/
```

and inspect the Webpack dev-server logs when the request arrives. That will help distinguish a `localhost`/IPv4/IPv6 binding issue, Host-header/server configuration, or the server actively resetting the Node connection.

So the timeout syntax is definitely wrong in the example; the actual socket reset needs the server/network diagnostics above if it remains afterward.

Please mark this answer as accepted if it helped, thank you.
